pkg/networkmanager/ip-settings.jsx

```diff
@@ -130,7 +130,25 @@ export const IpSettingsDialog = ({ topic, connection, dev, settings }) => {
 
         return false;
     };
-
+    const addressToMask = (address) => {
+        const addSplited = address.split('.');
+        if (addSplited.length !== 4)
+            return "";
+        if (addSplited[0] >= 0 && addSplited[0] <= 127) {
+            return "255.0.0.0";
+        } else if (addSplited[0] >= 128 && addSplited[0] <= 191) {
+            return "255.255.0.0";
+        } else if (addSplited[0] <= 192 && addSplited[0] <= 223) {
+            return "255.255.255.0";
+        }
+    };
+    const addressToGateway = (address, gateway) => {
+        const addSplited = address.split('.');
+        if (addSplited.length !== 4)
+            return "";
+        addSplited[3] = (address === gateway) ? (addSplited[3] === "1") ? "254" : "1" : (addSplited[3] === "1") ? "254" : "1";
```

Comment:
The outer ? and : branches compute exactly the same, so the `address === gateway` check does not make any difference. It's also hard to read the nested ?: , please break this up into three lines and enclose the inner ones with ().

How can address == gateway even happen? By definition the gateway is *not* your current machine, and you can't assign the same IP. This case should just be invalid, i.e. the helper should just return `null` then.